### ISSUE TITLE
Services need versioning

### ISSUE THREAD

**Issue by mellybelly** [monarch-initiative/monarch-legacy#299]
The services need version numbers, or can be released with same version as the whole app as part of our first official 1.0 release. Can the version information be added in response to service calls?

**Comment by kltm** [comment]
Is this API version or data version?

**Comment by cmungall** [comment]
I think we want both.

**Comment by cmungall** [comment]
Note this URL has the API version

/admin/introspect

But we should probably include this on a per-call basis

**Comment by mellybelly** [comment]
Yes exactly - need both.

**Comment by kltm** [comment]
Version aren't particularly useful unless you have a method of choosing--are you going to be archiving data and API versions so that they can be selected on the fly? And is the API independent of the data underneath?

**Comment by nlwashington** [comment]
yes, there are data versions and service versions.  in the case of the sufficiency score, the time of access will reflect (1) an algorithm for computing the score and (2) the data loaded into the owlsim server to feed into the algorithm.  the timestamp of access will tell you both, but of course isn't "constant".  what do you envision seeing, @mellybelly?

**Comment by cmungall** [comment]
What about ontology versions?

owlsim faithfully produces all ontology header info in json
http://owlsim.crbs.ucsd.edu/

this does look odd. Currently the makefile that initiates owlsim merges all ontologies into a single owl file. The original ontology headers are preserved, but as comments (they can't be preserved as ontology headers on the origin ontology, by the definition of an ontology merge). Not sure why the versionIRI for hp.owl isn't showing up there..

**Comment by mellybelly** [comment]
moving this to current release, please decide what is reasonable to implement wrt versioning. 
I guess if I had a result from Monarch, I would like to know what version of the data, what version of the software, and what version of the ontologies were used. Its not likely that someone could reproduce our data in the future given the magnitude of the architecture, but they can at least provide the best provenance possible so that if data/algorithms/ontologies change in the future they might know why.
